### PR TITLE
docs: update ceph-teardown doc

### LINKED ISSUE #7312: Docs: ceph-teardown.md outdated and ignores cleanup-policy technique
**Is this a bug report or feature request?**
* Bug Report (Docs)

**Deviation from expected behavior:**
[ceph-teardown.md](https://github.com/rook/rook/blob/master/Documentation/ceph-teardown.md) describes a painfully manual and outdated technique for cleaning disks.

**Expected behavior:**
Rook Docs, even early tutorials/walkthroughs, should include the latest, safest, idiomatic techniques.

**How to reproduce it (minimal and precise):**
1. Read [ceph-teardown.md](https://github.com/rook/rook/blob/master/Documentation/ceph-teardown.md)
1. Compare to [cleanup-policy feature](https://rook.io/docs/rook/v1.5/ceph-cluster-crd.html#cleanup-policy)
1. Observe that teardown.md instructs more dangerous and manual steps than are required.

### PR DESCRIPTION
**Pull request by sp98**
update ceph teardown doc to include cleanupPolicy which enables automatic clean up of the dataDirHostPath and the drives

Signed-off-by: Santosh Pillai <sapillai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #7312

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: Documentation/ceph-teardown.md [hunk 1]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
 kubectl delete storageclass csi-cephfs
 ```
 
-## Delete the CephCluster CRD
-
 After those block and file resources have been cleaned up, you can then delete your Rook cluster. This is important to delete **before removing the Rook operator and agent or else resources may not be cleaned up properly**.
 
+## Automatic Cleanup
+
+Edit the cephCluster and add the `cleanupPolicy`
```
Comment (leseb) [inline Documentation/ceph-teardown.md:39]:
```suggestion
Edit the `CephCluster` and add the `cleanupPolicy`
```

File: Documentation/ceph-teardown.md [hunk 2]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
...
+
+WARNING: DATA WILL BE PERMANENTLY DELETED AFTER DELETING THE `cephCluster` CRD WITH `cleanupPolicy`.
```
Comment (leseb) [inline Documentation/ceph-teardown.md:41]:
```suggestion
WARNING: DATA WILL BE PERMANENTLY DELETED AFTER DELETING THE `CephCluster` CR WITH `cleanupPolicy`.
```

File: Documentation/ceph-teardown.md [hunk 3]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
...
+
+```console
+kubectl -n rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
+```
+
+Once the cleanup policy is enabled, any new configuration changes in the cephCluster will be blocked. Nothing will happen unitl the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted.
```
Comment (leseb) [inline Documentation/ceph-teardown.md:47]:
```suggestion
Once the cleanup policy is enabled, any new configuration changes in the CephCluster will be blocked. Nothing will happen unitl the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted.
```

**Review by leseb — CHANGES_REQUESTED** [review]

File: Documentation/ceph-teardown.md [hunk 4]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
...
+
+Edit the `CephCluster` and add the `cleanupPolicy`
+
+WARNING: DATA WILL BE PERMANENTLY DELETED AFTER DELETING THE `CephCluster` CR WITH `cleanupPolicy`.
+
+```console
+kubectl -n rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
+```
+
+Once the cleanup policy is enabled, any new configuration changes in the CephCluster will be blocked. Nothing will happen unitl the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted.
```
Comment (travisn) [inline Documentation/ceph-teardown.md:47]:
```suggestion
Once the cleanup policy is enabled, any new configuration changes in the CephCluster will be blocked. Nothing will happen until the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted if needed.
```

File: Documentation/ceph-teardown.md [hunk 5]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
 kubectl delete storageclass csi-cephfs
 ```
 
-## Delete the CephCluster CRD
-
 After those block and file resources have been cleaned up, you can then delete your Rook cluster. This is important to delete **before removing the Rook operator and agent or else resources may not be cleaned up properly**.
 
+## Automatic Cleanup
```
Comment (travisn) [inline Documentation/ceph-teardown.md:37]: How about if we just keep this in the same section? We're just recommending to use the automatic cleanup by default now.
```suggestion
## Delete the CephCluster CR
```

File: Documentation/ceph-teardown.md [hunk 6]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
...
+
+```console
+kubectl -n rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
+```
+
+Once the cleanup policy is enabled, any new configuration changes in the CephCluster will be blocked. Nothing will happen unitl the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted.
+
+Checkout more details about the `cleanupPolicy` [here](https://rook.io/docs/rook/master/ceph-cluster-crd.html#cleanup-policy)
+
+## Delete the CephCluster CRD
```
Comment (travisn) [inline Documentation/ceph-teardown.md:51]: no need for this header if the other is renamed above

File: Documentation/ceph-teardown.md [hunk 7]
```diff
@@ -32,10 +32,24 @@ kubectl delete -f csi/cephfs/kube-registry.yaml
...
+
+WARNING: DATA WILL BE PERMANENTLY DELETED AFTER DELETING THE `CephCluster` CR WITH `cleanupPolicy`.
+
+```console
+kubectl -n rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
+```
+
+Once the cleanup policy is enabled, any new configuration changes in the CephCluster will be blocked. Nothing will happen unitl the deletion of the CR is requested, so this `cleanupPolicy` change can still be reverted.
+
+Checkout more details about the `cleanupPolicy` [here](https://rook.io/docs/rook/master/ceph-cluster-crd.html#cleanup-policy)
```
Comment (travisn) [inline Documentation/ceph-teardown.md:49]: Actually just refer to the .md files in the links, and the build process replaces with the correct html and branch link
```suggestion
Checkout more details about the `cleanupPolicy` [here](ceph-cluster-crd.md#cleanup-policy)
```

File: Documentation/ceph-teardown.md [hunk 8]
```diff
@@ -46,6 +60,12 @@ Verify that the cluster CRD has been deleted before continuing to the next step.
 kubectl -n rook-ceph get cephcluster
 ```
 
+If the `cleanupPolicy` was applied, then wait for the `rook-ceph-cleanup` jobs to be completed on all the nodes.
+These jobs :
```
Comment (travisn) [inline Documentation/ceph-teardown.md:64]:
```suggestion
These jobs will perform the following operations:
```

File: Documentation/ceph-teardown.md [hunk 9]
```diff
@@ -46,6 +60,12 @@ Verify that the cluster CRD has been deleted before continuing to the next step.
...
+- Clean up the `/var/lib/rook` (or the path specified by the `dataDirHostPath`) on all the nodes
```
Comment (travisn) [inline Documentation/ceph-teardown.md:65]:
```suggestion
- Delete the directory `/var/lib/rook` (or the path specified by the `dataDirHostPath`) on all the nodes
```

File: Documentation/ceph-teardown.md [hunk 10]
```diff
@@ -46,6 +60,12 @@ Verify that the cluster CRD has been deleted before continuing to the next step.
...
+- Delete the data on the drives on all the nodes
```
Comment (travisn) [inline Documentation/ceph-teardown.md:66]:
```suggestion
- Wipe the data on the drives on all the nodes where OSDs were running in this cluster
```

File: Documentation/ceph-teardown.md [hunk 11]
```diff
@@ -46,6 +60,12 @@ Verify that the cluster CRD has been deleted before continuing to the next step.
...
+
+Note: The cleanup jobs might not start if the resources created on top of Rook Cluster are not deleted completely. [See](https://github.com/rook/rook/blob/master/Documentation/ceph-teardown.md#delete-the-block-and-file-artifacts)
```
Comment (travisn) [inline Documentation/ceph-teardown.md:68]:
```suggestion
Note: The cleanup jobs might not start if the resources created on top of Rook Cluster are not deleted completely. [See](ceph-teardown.md#delete-the-block-and-file-artifacts)

```

File: Documentation/ceph-teardown.md [hunk 12]
```diff
@@ -57,6 +77,8 @@ kubectl delete -f common.yaml
 kubectl delete -f crds.yaml
 ```
 
+If the `cleanupPolicy` was applied and the cleanup jobs have completed on all the nodes, then the cluster tear down has been successful. If you have skipped adding the `cleanupPolicy` then follow the manual steps mentioned below to teardown the cluster.
```
Comment (travisn) [inline Documentation/ceph-teardown.md:80]:
```suggestion
If the `cleanupPolicy` was applied and the cleanup jobs have completed on all the nodes, then the cluster tear down has been successful. If you skipped adding the `cleanupPolicy` then follow the manual steps mentioned below to tear down the cluster.
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/ceph-teardown.md [hunk 13]
```diff
@@ -46,6 +59,12 @@ Verify that the cluster CRD has been deleted before continuing to the next step.
 kubectl -n rook-ceph get cephcluster
 ```
 
+If the `cleanupPolicy` was applied, then wait for the `rook-ceph-cleanup` jobs to be completed on all the nodes.
+These jobs will perform the following operations:
+- Delete the directory `/var/lib/rook` (or the path specified by the `dataDirHostPath`) on all the nodes
+- Wipe the data on the drives on all the nodes where OSDs were running in this cluster
+
+Note: The cleanup jobs might not start if the resources created on top of Rook Cluster are not deleted completely. [See](ceph-teardown.md#delete-the-block-and-file-artifacts)
```
Comment (travisn) [inline Documentation/ceph-teardown.md:67]: nit: add a newline

**Review by travisn — CHANGES_REQUESTED** [review]
one more tiny suggestion...

**Review by leseb — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**